### ISSUE TITLE
[Dark Mode] Button with FlatStyle 'Standard' is not in dark mode on 100% dpi after enabled SystemColorMode.Dark

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#11949]
### .NET version

9.0.0-rc.1.24418.2 + private dlls built from winforms repo main branch

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

This is a new feature Dark Mode: https://github.com/dotnet/winforms/pull/11857

### Issue description

On 100% DPI, Button with FlatStyle 'Standard' is not in dark mode after enabled SystemColorMode.Dark, it cannot repro when monitor's dpi is greater than 100%.

on 100%DPI:
<img width="61" alt="image" src="https://github.com/user-attachments/assets/5f344eb3-c2ec-45ea-8c0f-21915133bdb2">

on 150%DPI:
<img width="87" alt="image" src="https://github.com/user-attachments/assets/132bba48-8210-42a9-88f6-fb9b80494859">

### Steps to reproduce

1. Create a WinForms .NET application with button control, set button's FlatStyle == Standard
2. Enable DarkMode by Application.SetColorMode(SystemColorMode.Dark) in Program.cs file
3. Run the application to observe

**Comment by gerhardol** [comment]
A fix for this (and similar) issues is to set the back color when loading the form

In the (base) form constructor

```
Load += (s, e) => ((Form)s!).FixVisualStyle();
```

```
public static class ThemeFix
{
    private static readonly ConditionalWeakTable<IWin32Window, IWin32Window> AlreadyFixedControls =
        [];

    private static readonly ConditionalWeakTable<IWin32Window, IWin32Window> AlreadyFixedContextMenuOwners =
        [];

    public static ThemeSettings ThemeSettings { private get; set; } = ThemeSettings.Default;

    public static void FixVisualStyle(this Control container)
    {
        container.DescendantsToFix<Button>()
            .ForEach(SetupButton);
    }

    private static IEnumerable<TControl> DescendantsToFix<TControl>(this Control c)
        where TControl : Control
    {
        return c.FindDescendantsOfType<TControl>(SkipThemeAware)
            .Where(control => TryAddToWeakTable(control, AlreadyFixedControls));
    }

    private static bool SkipThemeAware(Control c) =>
        c.GetType().GetCustomAttribute<ThemeAwareAttribute>() is not null;
        private static void SetupButton(this Button button)
        {
            // .net9 fix for https://github.com/dotnet/winforms/issues/11949 (only supposed to occur for 100%)
#pragma warning disable WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
            if (Application.IsDarkModeEnabled && button.FlatStyle == FlatStyle.Standard)
            {
                button.TouchBackColor();
            }
#pragma warning restore WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
        }


        private static void TouchBackColor(this Control c)
        {
            c.BackColor = c.BackColor;
        }

        private static void TouchForeColor(this Control c)
        {
            c.ForeColor = c.ForeColor;
        }

        private static bool TryAddToWeakTable(IWin32Window element, ConditionalWeakTable<IWin32Window, IWin32Window> weakTable)
        {
            if (weakTable.TryGetValue(element, out _))
            {
                return false;
            }

            weakTable.Add(element, element);
            return true;
        }
}
```

**Comment by AngeloCresta** [comment]
Just my 2 cents:

![Image](https://github.com/user-attachments/assets/81edefeb-1228-4fb1-a5cf-5cc1d2050675)

```
            // 
            // btnClear
            // 
            btnClear.Image = Properties.Resources.TrashCan
            ...
            btnClear.FlatStyle = System.Windows.Forms.FlatStyle.Popup;
            // 
            // btnSaveData
            // 
            btnSaveData.Image = Properties.Resources.FileSaveIcon;
            ...
            btnSaveData.FlatStyle = System.Windows.Forms.FlatStyle.System;
            // 
            // btnGenerate
            // 
            btnGenerate.Image = (System.Drawing.Image)resources.GetObject("btnGenerate.Image");
            ...
            btnGenerate.FlatStyle = System.Windows.Forms.FlatStyle.Standard;
```
"System.Windows.Forms.FlatStyle.Popup" --> ok colors, ok image
"System.Windows.Forms.FlatStyle.System" -->ok colors, No image
"System.Windows.Forms.FlatStyle.Standard"--> Nok colors, ok image

All with "Popup":

![Image](https://github.com/user-attachments/assets/6d0ef382-9144-4588-9a1a-48a5b27e0819)

Angelo

**Comment by Olina-Zhang** [comment]
Verified it in the latest .NET 10 SDK build: 10.0.100-preview.6.25317.107, it was fixed:

![Image](https://github.com/user-attachments/assets/65414813-aa5d-4170-9f00-f943ec8f54c2)

**Comment by Olina-Zhang** [comment]
Tested it in the .NET 10 Preview 6 test pass build, it has same result as above.
